pkg/utils/flow/flow.go

```diff
@@ -167,14 +166,9 @@ func newExecution(flow *Flow, opts Opts) *execution {
 		all.Insert(name)
 	}
 
-	var log interface{}
-	switch logger := opts.Logger.(type) {
-	case logrus.FieldLogger:
-		log = logger.WithField(logKeyFlow, flow.name)
-	case logr.Logger:
-		log = logger.WithValues(logKeyFlow, flow.name)
-	default:
-		log = logf.Log.WithName("flow").WithValues(logKeyFlow, flow.name)
+	log := logf.Log.WithName("flow").WithValues(logKeyFlow, flow.name)
+	if opts.Log != (logr.Logger{}) {
```

Comment:
I think, the proper way would be:
```suggestion
	if opts.Log.GetSink() != nil {
```